pgsql/client.py

```diff
@@ -157,14 +170,20 @@ def extensions(self) -> List[str]:
         # appdata = self.relation.data[self.unit.app]
         appdata = _get_pgsql_leader_data().get(self.relation.id, {})
         sext = appdata.get("extensions") or ""
-        return list(sext.split(","))
+        return list(e for e in sext.split(",") if e)
 
     @extensions.setter
     def extensions(self, extensions: Iterable[str]) -> None:
+        if extensions is None:
+            extensions = []
         sext = ",".join(sorted(extensions))
         self.relation.data[self._local_unit.app]["extensions"] = sext
         # Deprecated, per PostgreSQLClient._mirror_appdata()
         self.relation.data[self._local_unit]["extensions"] = sext  # Deprecated, should be app reldata
+        # Inform our peers, since they can't read the appdata
+        d = _get_pgsql_leader_data()
+        d.setdefault(self.relation.id, {})["extensions"] = sext
```

Comment:
I've tried to run something like

`$ juju run --application mattermost "leader-get --format=yaml"`

To see how the data would look like here, to no avail.
It would be nice to have a comment with the kind of data expected.